src/iso15118/io/connection_openssl.cpp

```diff
@@ -107,7 +107,7 @@ static SSL_CTX* init_ssl(const config::SSLConfig& ssl_config) {
                 return;
             }
             const auto result = key_logging_server.send(line);
-            if (result < 0 or (result >= 0 and static_cast<size_t>(result) != strlen(line))) {
+            if (result < 0 or cmp_equal(result, strlen(line))) {
```

Comment:
shouldn't this be just `if (not cmp_equal(result, strlen(line)))`?

Reply:
I think `result` will be `-1` if an error has occurred during `send()`. And the cmp_equal does not check that

Reply:
yes, but still `-1` is not equal to `strlen(line)`

Reply:
Yeaah you are correct. I'll change that